specs/customizability.md

```diff
@@ -0,0 +1,350 @@
+
+- `fendermint/actors/customsyscall/Cargo.toml`: The package for your new actor and all its dependencies
+- `fendermint/actors/Cargo.toml`: Add your new actor as a Wasm target
+- `fendermint/actors/build.rs`: Include your new actor in the `ACTORS` array so it will get included in the bundle.
+- `fendermint/actors/src/manifest.rs`: Add your new actor in the `REQUIRED_ACTORS` array so we can confirm it was correctly bundled on IPC startup
+- `fendermint/vm/actor_interface/src/customsyscall.rs`: A macro which assigns an ID to your new actor and declares constants for accessing it by ID and Address
+- `fendermint/vm/actor_interface/src/lib.rs`: export the constants to IPC
+</aside>
+
+### **7. Load and deploy actor at genesis**
+
+We have so far created a new kernel and syscall, switched IPC to use that kernel and created an actor which calls the new syscall. However, in order to call this actor in IPC, we must load it from the custom_actors_bundle.
+
+To do this open `fendermint/vm/interpreter/src/fvm/genesis.rs` file and in the `init` function add our customsyscall actor right after creating the `chainmetadata` actor:
+
+```rust
+// Initialize the customsyscall actor which gives an example of calling a custom syscall
+state
+    .create_custom_actor(
+        fendermint_actor_customsyscall::CUSTOMSYSCALL_ACTOR_NAME,
+        customsyscall::CUSTOMSYSCALL_ACTOR_ID,
+        &EMPTY_ARR,
+        TokenAmount::zero(),
+        None,
+    )
+    .context("failed to create customsyscall actor")?;
+```
+
+[fendermint/vm/interpreter/src/fvm/genesis.rs#L251](https://github.com/consensus-shipyard/ipc/blob/98497363a10e08236325e6d5c52755b9fcd52958/fendermint/vm/interpreter/src/fvm/genesis.rs#L251)
+
+Your actor has now been deployed and we should be able to send it messages!
+
+# Genesis Parameters
+
+The contents of the `genesis.json` file are essentially the `Genesis` structure defined in the [genesis](https://github.com/consensus-shipyard/ipc/tree/specs/fendermint/vm/genesis) crate. It has the following properties:
+
+- `chain_name` is an arbitrary name for the chain, which will be hashed to become the numeric `chain_id`; in the context of IPC subnets the `chain_name` is expected to be the textual representation of the `SubnetID`
+- `timestamp` is the number of seconds since UNIX epoch
+- `network_version` is used by the FVM to select gas pricing policy
+- `base_fee` is measured in *atto* and represents the base price for gas
+- `power_scale` is the number of decimals to take into account from the FIL token collateral balance when converting it into voting power expressed as `u64`, which is what CometBFT expects. For example if the scale is 0 then every 1 FIL gives 1 voting power, if it’s 3 then every 0.001 FIL does, and if it’s -1 then every 10 FIL. The power is rounded upwards, so that we don’t end up with 0 power, which would be rejected by CometBFT; for example if the scale is 1, then both 1.1 FIL and 1.9 FIL give 2 power.
```

Comment:
maybe mention that it only works for the `Collateral` (ie PoS) mode?

Reply:
So... as we've noted recently, this applies in `Federated` mode too.